Firestore/core/src/firebase/firestore/nanopb/nanopb_string.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_NANOPB_NANOPB_STRING_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_NANOPB_NANOPB_STRING_H_
+
+#include <pb.h>
+
+#include <cstdlib>
+#include <functional>
+#include <string>
+#include <utility>
+
+#include "Firestore/core/src/firebase/firestore/util/comparison.h"
+#include "absl/strings/string_view.h"
+
+namespace firebase {
+namespace firestore {
+namespace nanopb {
+
+class String : public util::Comparable<String> {
```

Comment:
docstrings might be nice. Specifically, something mentioning that the ctor copies it's arguments.